### ISSUE TITLE
[Bug] fa2 prefill with an fp8 KV cache is wrong for head_dim_qk=192 / head_dim_vo=128

### ISSUE THREAD

**Issue by SamMausberg** [flashinfer-ai/flashinfer#4976]
### Before submitting

- [x] I have searched existing issues and did not find a solution or an existing report.

### Environment

GH200 (SM90), CUDA 12.8, torch 2.14.0+cu130, FlashInfer main at df8b5c17 (0.6.18).

### Bug description

`BatchPrefillWithRaggedKVCacheWrapper` with `backend="fa2"`, `head_dim_qk=192`, `head_dim_vo=128` (the DeepSeek prefill head-dim pair) and an fp8 KV cache returns wrong outputs for every row and every output dimension. The same call with a 16-bit KV cache is correct, and fp8 KV with `head_dim_qk == head_dim_vo` (64 / 128 / 256) is correct, so the defect is specific to the mixed-precision path with unequal head dims. Both `float8_e4m3fn` and `float8_e5m2`, both `float16` and `bfloat16` queries, causal and non-causal are affected.

Max abs error against an fp32 reference over the dequantized cache (99 tokens, 16 heads, randn inputs):

| query | KV cache | causal | max err |
|---|---|---|---|
| fp16 | e4m3 | no | 1.19 |
| fp16 | e5m2 | no | 1.49 |
| bf16 | e4m3 | no | 1.41 |
| bf16 | e5m2 | no | 1.34 |
| fp16 | e4m3 | yes | 2.49 |
| fp16 | fp16 | no | 0.0005 |

### Minimal reproduction

```python
import math, torch, flashinfer

torch.manual_seed(0)
seq_len, H, DQK, DVO = 99, 16, 192, 128
q_dtype, kv_dtype = torch.float16, torch.float8_e4m3fn
ws = torch.empty(128 << 20, dtype=torch.uint8, device="cuda")
q = torch.randn(seq_len, H, DQK, dtype=q_dtype, device="cuda")
k = torch.randn(seq_len, H, DQK, dtype=q_dtype, device="cuda").to(kv_dtype)
v = torch.randn(seq_len, H, DVO, dtype=q_dtype, device="cuda").to(kv_dtype)
indptr = torch.tensor([0, seq_len], dtype=torch.int32, device="cuda")

w = flashinfer.BatchPrefillWithRaggedKVCacheWrapper(ws, "NHD", backend="fa2")
w.plan(indptr, indptr, H, H, DQK, head_dim_vo=DVO, causal=False,
       q_data_type=q_dtype, kv_data_type=kv_dtype)
o = w.run(q, k, v).float()

s = torch.einsum("qhd,khd->hqk", q.float(), k.float()) / math.sqrt(DQK)
ref = torch.einsum("hqk,khd->qhd", torch.softmax(s, -1), v.float())
print("max err", (o - ref).abs().max().item())   # 1.19; 0.0005 with kv_dtype=torch.float16
```

### Additional context

Found while adding fa3 support for a 16-bit query with an fp8 KV cache (#3327); the fa3 kernel matches the reference on this head-dim pair, so the new test compares against fa3 rather than fa2 for it.
